File: Bit Manipulation/conversion.java
public class conversion{

  static int convers(long a, long b){
    int count=0;
    while(a!=0||b!=0){
      if((a&1)!=(b&1)) count++;
      a>>=1;
      b>>=1;
    }
    return count;
  }

  public static void main(String[] args){
    long a=29;
    long b=15;
    System.out.println(convers(a,b));
  }

}

File: Bit Manipulation/conversion.cpp
#include <stdio.h>
#include <iostream>

using namespace std;

unsigned long convers(unsigned long a, unsigned long b){
  unsigned long count=0;
  while(a!=0||b!=0){
    // printf("%lu %lu\n",a,b);
    if((a&1)!=(b&1)) count++;
    // printf("count: %lu\n",count);
    a>>=1;
    b>>=1;
  }
  return count;
}

int main(){
  unsigned long a=29;
  unsigned long b=15;

  printf("%lu\n",convers(a,b));

}
